cli/cmd/project/connect_github.go

```diff
@@ -0,0 +1,662 @@
+package project
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing"
+	githttp "github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/google/go-github/v50/github"
+	"github.com/rilldata/rill/admin/pkg/urlutil"
+	"github.com/rilldata/rill/cli/cmd/auth"
+	"github.com/rilldata/rill/cli/cmd/org"
+	"github.com/rilldata/rill/cli/pkg/browser"
+	"github.com/rilldata/rill/cli/pkg/cmdutil"
+	"github.com/rilldata/rill/cli/pkg/dotrillcloud"
+	"github.com/rilldata/rill/cli/pkg/gitutil"
+	"github.com/rilldata/rill/cli/pkg/printer"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/rilldata/rill/runtime/pkg/activity"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	pollTimeout  = 10 * time.Minute
+	pollInterval = 5 * time.Second
+)
+
+func GitPushCmd(ch *cmdutil.Helper) *cobra.Command {
+	opts := &DeployOpts{}
+
+	deployCmd := &cobra.Command{
+		Use:   "connect-github",
+		Short: "Deploy project to Rill Cloud by pulling project files from a git repository",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return ConnectGithubFlow(cmd.Context(), ch, opts)
+		},
+	}
+
+	deployCmd.Flags().SortFlags = false
+	deployCmd.Flags().StringVar(&opts.GitPath, "path", ".", "Path to project repository (default: current directory)") // This can also be a remote .git URL (undocumented feature)
+	deployCmd.Flags().StringVar(&opts.SubPath, "subpath", "", "Relative path to project in the repository (for monorepos)")
+	deployCmd.Flags().StringVar(&opts.RemoteName, "remote", "", "Remote name (default: first Git remote)")
+	deployCmd.Flags().StringVar(&ch.Org, "org", ch.Org, "Org to deploy project in")
+	deployCmd.Flags().StringVar(&opts.Name, "project", "", "Project name (default: Git repo name)")
+	deployCmd.Flags().StringVar(&opts.Description, "description", "", "Project description")
+	deployCmd.Flags().BoolVar(&opts.Public, "public", false, "Make dashboards publicly accessible")
+	deployCmd.Flags().StringVar(&opts.Provisioner, "provisioner", "", "Project provisioner")
+	deployCmd.Flags().StringVar(&opts.ProdVersion, "prod-version", "latest", "Rill version (default: the latest release version)")
+	deployCmd.Flags().StringVar(&opts.ProdBranch, "prod-branch", "", "Git branch to deploy from (default: the default Git branch)")
+	deployCmd.Flags().IntVar(&opts.Slots, "prod-slots", 2, "Slots to allocate for production deployments")
+	if !ch.IsDev() {
+		if err := deployCmd.Flags().MarkHidden("prod-slots"); err != nil {
+			panic(err)
+		}
+	}
+
+	// 2024-07-27: We have deprecated configuration of the OLAP DB using flags in favor of using rill.yaml.
```

Comment:
This date is `2024-02-19` on main, why change it? (If anything, we can probably remove these options now.)

Reply:
Removing is probably better if it is not used, else I will change the date back.